internal/provider/resource_gitlab_group.go

```diff
@@ -369,9 +368,40 @@ func resourceGitlabGroupUpdate(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 
+	if d.HasChange("parent_id") {
+		diagnostic := transferSubGroup(d, client)
+		if diagnostic.HasError() {
+			return diagnostic
+		}
+	}
+
 	return resourceGitlabGroupRead(ctx, d, meta)
 }
 
+func transferSubGroup(d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
(non-blocking): I think it would make sense to already pass the GitLab client and attributes needed directly to this function. Also, return an `error` instead of `Diagnostics` - we don't use the warning feature anyways.

WDYT?

Reply:
no emotions, I will fix it...